File: Task-03/Python/Sub-Task4.py
def pyramid(rows):
    pattern = []

    for i in range(rows):
        line = " " * (rows - i - 1) + "* " * (i + 1)
        pattern.append(line)
    for j in range(rows - 1, 0, -1):
        line = " " * (rows - j) + "* " * j
        pattern.append(line)
    return pattern
def main():
   with open('input.txt', 'r') as infile:
        rows = int(infile.read().strip())

   pattern = pyramid(rows)

   with open('output.txt', 'w') as outfile:
        for line in pattern:
            outfile.write(line + '\n')
main()

File: Task-03/Java/Sub-Task4.java
import java.io.FileReader;
import java.io.FileWriter;
import java.util.Scanner;

public class SubTask4 {
    public static void main(String[] args) {
        FileReader reader = new FileReader("input.txt");
        Scanner fileScanner = new Scanner(reader);
        int rows = fileScanner.nextInt();
        fileScanner.close();

        FileWriter writer = new FileWriter("output.txt");

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < rows - i - 1; j++) {
                writer.write(" ");
            }
            for (int j = 0; j <= i; j++) {
                writer.write("* ");
            }
            writer.write("\n");
        }
        for (int i = 0; i < rows - 1; i++) {
            for (int j = 0; j <= i; j++) {
                writer.write(" ");
            }
            for (int j = 0; j < rows - i - 1; j++) {
                writer.write("* ");
            }
            writer.write("\n");
        }

        writer.close();
    }
}
